File: C++/0070. 循环小数.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
int gcd(int a, int b){
    while(a%b!=0){
        a %= b;
        if(a<b) swap(a,b);
    }
    return b;
}
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    string s;
    while(cin>>s){
        if(s=="0"){
            return 0;
        }
        int p = 1;
        int l =s.size();
        while(l-->0){
            p *= 10;
        }
        p--;
        int n = stoi(s);
        int x = gcd(n,p);
        cout<<n/x<<'/'<<p/x<<endl;
    }
    return 0;
}

File: Python/0070. 循环小数.py
# If you need to import additional packages or classes, please import here.
def gcd(a,b):
    while a%b:
        a = a%b
        if a<b:
            a,b = b,a
    return b
def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n = input()
            l = len(n)
            n = int(n)
            if n == 0:
                break
            p = pow(10,l)-1
            x = gcd(p,n)
            print(str(n//x)+'/'+str(p//x))
        
        except EOFError:
            break
    
if __name__ == "__main__":
    func()
